### ISSUE TITLE
Inspecting multitask loss in Graph instances

### ISSUE THREAD

**Issue by qxcv** [fchollet/keras#1772]
When training or validating a `Graph` model with two objectives, it's frustrating to be unable to see both losses at once. Right now, for instance, I'm trying to train a network to regress and classify at the same time, but it's impossible to balance the regression and classification losses because I can't see how large they are. For one illustration of how the balance parameter in a multitask loss can impact accuracy, look at Table 9 in [Faster R-CNN: Towards Real-Time Object Detection with Region Proposal Networks](http://arxiv.org/pdf/1506.01497v3.pdf) (loss terms and relative importance of λ explained below Eqn. 1 on p5).
#580 presents a workaround, but that workaround is slow (especially if you want to see each loss on every minibatch!) and error-prone. Ideally `train_on_batch`, `validate_on_batch`, etc. should give a dictionary of losses like

```
{
  'regressor_loss': 23.5185,
  'classifier_loss': 3.9618
}
```

The return value and debugging output of `.fit()` and other convenience methods would have to be similarly modified. This probably breaks a lot of old code, so it may be necessary to introduce a flag analogous to `accuracy={True,False}` which indicates whether all losses should be returned or just one.

Kind-of related: #1510.

**Comment by qxcv** [comment]
Just remembered this issue. Turns out multitask loss is really easy to monitor with new `Model` API, e.g.

```python
from keras.models import Model
from keras.layers import Input, Dense, Activation
import numpy as np
import matplotlib.pyplot as plt

input = Input(shape=(1,))
x = Dense(128)(input)
x = Activation('sigmoid')(x)
x = Dense(128)(x)
x = Activation('relu')(x)
out_mse = Dense(1, name='out_mse')(x)
out_mae = Dense(1, name='out_mae')(x)
m = Model(input=[input], output=[out_mse, out_mae])
m.compile(optimizer='rmsprop', loss={'out_mse': 'mse', 'out_mae': 'mae'})

X = (np.random.random((100000, 1)) - 0.5) * 4 * np.pi
y = np.cos(X)
y += 0.05 * np.random.randn(*y.shape)
# .fit() will display both objective values at once, by default
m.fit(X, {'out_mse': y, 'out_mae': y}, batch_size=1000)
# can also get values manually
total, mse_loss, mae_loss = m.train_on_batch(X, {'out_mse': y, 'out_mae': y})
print('MSE: %g, MAE: %g, total: %g' % (mse_loss, mae_loss, total))

Xv = np.linspace(-4 * np.pi, 4 * np.pi, 1000).reshape((-1, 1))
mse_pred, mae_pred = m.predict(Xv)
plt.plot(Xv.flatten(), mse_pred.flatten(), label='$\ell_2$')
plt.plot(Xv.flatten(), mae_pred.flatten(), label='$\ell_1$')
plt.legend()
plt.show()
```

Yay Keras!
